README.md

```diff
@@ -2,17 +2,71 @@
 
 ## About ##
 
-This is an attempt to build just another web-crawler, also called _Spider_. Its purpose is to get data from a website in a form of a list of objects. This data can be then used to download linked files and place them in a database or a folder structure.
+Onigumo je web-crawler.
+
+Onigumo "prolézá" webové stránky či aplikace. Jejich obsah a případná metadata uloží do strukturované podoby, která je vhodná pro další strojové zpracování.
 
 ## Architecture ##
 
-### Building blocks ###
+Onigumo je rozděleno do tří na sebe vzájemně navazujících částí:
+
+* operator - řízení
+* downloader - stahování dat
+* parser - zpracování dat
+
+Diagram níže, znázorňuje vzájemnou součinnost těchto celků:
+
+```mermaid
+flowchart LR
+    START             -->           operator(OPERATOR)
+    operator    -- urls.txt --->    downloader(DOWNLOADER)
+    downloader  -- *.html --->      parser(PARSER)
+    parser      -- *.JSON -->       operator
+    operator           -->          MATERIALIZATION
+```
+
+### Operator ###
+
+Operator vytváří frontu URL adres určených ke stažení pro _downloader_. Za přidávání URL adres určených ke zpracování je zodpovědná aplikace. Nové URL adresy aplikace získává z naparsované podoby dat, kterou vytváří _parser_.
+
+Činnost _operatoru_ se skládá z:
+
+1. inicializace práce Oniguma na dané aplikaci
+2. kontroly stavu zpracovaných a nezpracovaných URL adres z výstupu _operatoru_, popř. dle zapsaných souborů
```

Comment:
```suggestion
```

This is not the operator’s job. It only checks the parsed files for further actions, that is, more URLs to download.